File: src/main/java/me/wuzzyxy/dynamicmarket/placeholders/SellPriceExpansion.java
package me.wuzzyxy.dynamicmarket.placeholders;

import me.clip.placeholderapi.expansion.PlaceholderExpansion;
import me.wuzzyxy.dynamicmarket.DynamicMarket;
import me.wuzzyxy.dynamicmarket.items.MarketItem;
import me.wuzzyxy.dynamicmarket.market.MarketManager;
import me.wuzzyxy.dynamicmarket.market.PriceHandler;
import org.bukkit.OfflinePlayer;
import org.jetbrains.annotations.NotNull;

import java.text.DecimalFormat;

public class SellPriceExpansion extends PlaceholderExpansion {
    private final DynamicMarket plugin;
    private final MarketManager manager;
    private final PriceHandler priceHandler;
    DecimalFormat df;
    public SellPriceExpansion(DynamicMarket plugin, MarketManager manager, PriceHandler priceHandler) {
        this.plugin = plugin;
        this.manager = manager;
        this.priceHandler = priceHandler;

        df = new DecimalFormat("0.00");
    }

    @Override
    public @NotNull String getIdentifier() {
        return "DMSellPrice";
    }

    @Override
    public @NotNull String getAuthor() {
        return "Wuzzy";
    }

    @Override
    public @NotNull String getVersion() {
        return plugin.getDescription().getVersion();
    }
    @Override
    public boolean persist() {
        return true; //
    }

    public String onRequest(OfflinePlayer player, @NotNull String params) {
        String[] args = params.split(",");

        if (args.length != 2) return "DMSellPrice_Item,Amount" + args.length + " args provided";

        MarketItem item = manager.getPersistedItem(args[0]).orElse(null);
        if (item == null) return args[0]+ " not found";

        try {
            int amount = Integer.parseInt(args[1]);
            double price = priceHandler.getSellPrice(item, amount);
            return df.format(price);
        } catch (NumberFormatException e) {
            return "Invalid params";
        }
    }


}

File: src/main/kotlin/me/wuzzyxy/dynamicmarket/placeholders/SellPriceExpansion.kt
package me.wuzzyxy.dynamicmarket.placeholders

import me.clip.placeholderapi.expansion.PlaceholderExpansion
import me.wuzzyxy.dynamicmarket.DynamicMarket
import me.wuzzyxy.dynamicmarket.market.MarketManager
import me.wuzzyxy.dynamicmarket.market.PriceHandler
import org.bukkit.OfflinePlayer
import java.text.DecimalFormat

class SellPriceExpansion(
    private val plugin: DynamicMarket,
    private val manager: MarketManager,
    private val priceHandler: PriceHandler,
) : PlaceholderExpansion() {

    private val df = DecimalFormat("0.00")

    override fun getIdentifier(): String = "DMSellPrice"

    override fun getAuthor(): String = "Wuzzy"

    override fun getVersion(): String = plugin.description.version

    override fun persist(): Boolean = true

    override fun onRequest(player: OfflinePlayer?, params: String): String {
        val args = params.splitParams()

        if (args.size != 2) return "DMSellPrice_Item,Amount${args.size} args provided"

        val item = manager.getWorkingItem(args[0]) ?: return "${args[0]} not found"
        val amount = args[1].toIntOrNull() ?: return "Invalid params"

        return df.format(priceHandler.getSellPrice(item, amount))
    }
}
